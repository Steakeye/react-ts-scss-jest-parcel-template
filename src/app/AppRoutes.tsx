import { ExtendedRouteProps } from './AppRoutes.d';
import { Home } from './views/home/Home';
import { Auth } from './views/auth/Auth';
import { NotFound } from '~/app/views/not-found/NotFound';
import { ServerError } from '~/app/views/server-error/ServerError';

export const BASE_ROUTE = '/';
export const AUTH_ROUTE = '/auth';
export const ERROR_ROUTE = '/error';

const appRoutes: ExtendedRouteProps[] = [
  {
    path: BASE_ROUTE,
    component: Home,
    exact: true,
    name: 'Home View',
  },
  {
    path: AUTH_ROUTE,
    component: Auth,
    exact: true,
    name: 'Auth View',
  },
  {
    path: ERROR_ROUTE,
    component: ServerError,
    exact: true,
    name: 'Server Error View',
  },
  {
    path: '/*',
    component: NotFound,
    exact: true,
    name: 'Not Found View',
  },
];

export { appRoutes as AppRoutes };
