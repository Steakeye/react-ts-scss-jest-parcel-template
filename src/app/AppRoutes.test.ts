import { ExtendedRouteProps } from '~/app/AppRoutes.d';

jest.mock('./views/home/Home', () => ({
  Home: 'MockHomeView',
}));

jest.mock('./views/auth/Auth', () => ({
  Auth: 'MockAuthView',
}));

jest.mock('./views/server-error/ServerError', () => ({
  ServerError: 'MockServerErrorView',
}));

jest.mock('./views/not-found/NotFound', () => ({
  NotFound: 'MockNotFoundView',
}));

let AppRoutes: ExtendedRouteProps[];

beforeAll(async () => {
  ({ AppRoutes } = await import('./AppRoutes'));
});

describe('Application Routes - route configuration and route related helper functions', () => {
  describe('Application Route configuration', () => {
    it('App routes are configured in a list of routes', () => {
      expect(AppRoutes).toMatchSnapshot();
    });
  });
});
