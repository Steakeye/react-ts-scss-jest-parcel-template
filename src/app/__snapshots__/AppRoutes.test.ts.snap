// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application Routes - route configuration and route related helper functions Application Route configuration App routes are configured in a list of routes 1`] = `
Array [
  Object {
    "component": "MockHomeView",
    "exact": true,
    "name": "Home View",
    "path": "/",
  },
  Object {
    "component": "MockAuthView",
    "exact": true,
    "name": "Auth View",
    "path": "/auth",
  },
  Object {
    "component": "MockServerErrorView",
    "exact": true,
    "name": "Server Error View",
    "path": "/error",
  },
  Object {
    "component": "MockNotFoundView",
    "exact": true,
    "name": "Not Found View",
    "path": "/*",
  },
]
`;
